pkg/compactor/compactor.go

```diff
@@ -607,6 +613,8 @@ func (c *Compactor) compactUsers(ctx context.Context) {
 		if succeeded && compactionErrorCount == 0 {
```

Comment:
If we introduce an additional `interrupted` var then seems `succeeded` is unncessary anymore.

Reply:
We can actually do away with the error count as well and just have a `failed` boolean given we do not use the compactionErrorCount for anything than comparing with zero.  See latest updates.